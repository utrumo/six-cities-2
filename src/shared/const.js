const MAX_IMAGE_ON_OFFER_PAGE = 6;
const ASSETS_PATCH = `/`;

const OfferTypeToPresentName = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export {
  MAX_IMAGE_ON_OFFER_PAGE,
  ASSETS_PATCH,
  OfferTypeToPresentName
};
