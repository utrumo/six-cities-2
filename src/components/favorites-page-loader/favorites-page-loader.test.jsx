import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {MemoryRouter as Router} from 'react-router-dom';
import NameSpace from '../../store/name-spaces.js';
import {DEFAULT_NUMBER_VALUE, SortingVariants} from '../../shared/const.js';
import FavoritesPageLoader from './favorites-page-loader.jsx';
const mockStore = configureStore();

it(`FavoritesPageLoader should render favorites empty page if hasn't favorite offers`, () => {
  const offers = [
    {
      id: 1,
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13,
        },
      },
      previewImage: `/img/apartment-01.jpg`,
      images: [
        `/img/apartment-02.jpg`,
        `/img/room.jpg`,
        `/img/apartment-01.jpg`,
        `/img/studio-01.jpg`,
        `/img/apartment-03.jpg`,
        `/img/apartment-01.jpg`,
        `/img/apartment-01.jpg`,
      ],
      title: `Beautiful & luxurious apartment at great location`,
      description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
      location: {
        latitude: 52.364540000000005,
        longitude: 4.9019759999999994,
        zoom: 16,
      },
      isPremium: true,
      isFavorite: false,
      rating: 4.6,
      price: 120,
      type: `apartment`,
      bedrooms: 5,
      maxAdults: 3,
      goods: [
        `Washer`,
        `Laptop friendly workspace`,
        `Breakfast`,
        `Dishwasher`,
        `Fridge`,
        `Air conditioning`,
        `Baby seat`,
        `Towels`,
        `Coffee machine`,
      ],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`,
      },
    },
  ];
  const store = mockStore({
    [NameSpace.DATA]: {
      offers,
      offersReviews: [],
      currentLocation: ``,
      currentOfferId: DEFAULT_NUMBER_VALUE,
      sortOrder: SortingVariants.POPULAR,
    },
    [NameSpace.USER]: {
      isAuthorized: true,
      emailValidationError: ``,
      profile: {
        id: 1,
        email: `test@test.ru`,
        name: `test`,
        avatarUrl: `/static/avatar/2.jpg`,
        isPro: false,
      },
    },
  });
  const favoritePlaces = renderer.create(
      <Provider store={store}>
        <Router>
          <FavoritesPageLoader />
        </Router>
      </Provider>
  ).toJSON();
  expect(favoritePlaces).toMatchSnapshot();
});

it(`FavoritesPageLoader should render favorites page if has favorite offers`, () => {
  const offers = [
    {
      id: 1,
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13,
        },
      },
      previewImage: `/img/apartment-01.jpg`,
      images: [
        `/img/apartment-02.jpg`,
        `/img/room.jpg`,
        `/img/apartment-01.jpg`,
        `/img/studio-01.jpg`,
        `/img/apartment-03.jpg`,
        `/img/apartment-01.jpg`,
        `/img/apartment-01.jpg`,
      ],
      title: `Beautiful & luxurious apartment at great location`,
      description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
      location: {
        latitude: 52.364540000000005,
        longitude: 4.9019759999999994,
        zoom: 16,
      },
      isPremium: true,
      isFavorite: true,
      rating: 4.6,
      price: 120,
      type: `apartment`,
      bedrooms: 5,
      maxAdults: 3,
      goods: [
        `Washer`,
        `Laptop friendly workspace`,
        `Breakfast`,
        `Dishwasher`,
        `Fridge`,
        `Air conditioning`,
        `Baby seat`,
        `Towels`,
        `Coffee machine`,
      ],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`,
      },
    },
  ];
  const store = mockStore({
    [NameSpace.DATA]: {
      offers,
      offersReviews: [],
      currentLocation: ``,
      currentOfferId: DEFAULT_NUMBER_VALUE,
      sortOrder: SortingVariants.POPULAR,
    },
    [NameSpace.USER]: {
      isAuthorized: true,
      emailValidationError: ``,
      profile: {
        id: 1,
        email: `test@test.ru`,
        name: `test`,
        avatarUrl: `/static/avatar/2.jpg`,
        isPro: false,
      },
    },
  });
  const favoritePlaces = renderer.create(
      <Provider store={store}>
        <Router>
          <FavoritesPageLoader />
        </Router>
      </Provider>
  ).toJSON();
  expect(favoritePlaces).toMatchSnapshot();
});
