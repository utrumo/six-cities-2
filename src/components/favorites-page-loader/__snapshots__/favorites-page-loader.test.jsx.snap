// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPageLoader should render favorites empty page if hasn't favorite offers 1`] = `
<div
  className="page page--favorites-empty"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            aria-current="page"
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "cursor": "default",
              }
            }
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  test@test.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;

exports[`FavoritesPageLoader should render favorites page if has favorite offers 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            aria-current="page"
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "cursor": "default",
              }
            }
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  test@test.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <div
              className="places__list favorites__places"
            >
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/1"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="/img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        120
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      Beautiful & luxurious apartment at great location
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
